engine/src/main/java/org/pentaho/di/base/AbstractMeta.java

```diff
@@ -2175,4 +2174,7 @@ public void setMetaFileCache( IMetaFileCache metaFileCache ) {
     this.metaFileCache = metaFileCache;
   }
 
+  public List<DatabaseMeta> getLocalDbMetas() {
+    return new ArrayList<>( localDbMetas.values());
+  }
```

Comment:
don't the callers modify this? (which is also not recommended)

Reply:
localDbMetas is something I added. There is currently no setters for it. It gets loaded during loadXml(). We will need to handle the case when someone is updating the ktr/kjb or creating a new ktr/kjb.